dataland-external-storage/src/main/kotlin/org/dataland/datalandexternalstorage/services/DatabaseConnection.kt

```diff
@@ -0,0 +1,85 @@
+import org.slf4j.LoggerFactory
+import java.sql.Connection
+import java.sql.DriverManager
+import java.sql.PreparedStatement
+import java.util.Properties
+import java.util.UUID
+
+/**
+ * This object holds methods to establish a JDBC connection and do sql inserts into a database table
+ */
+object DatabaseConnection {
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * The method prepares and executes a sql statement to insert a pair of dataId data into an external database
+     * @param conn hold the connection details
+     * @param sqlStatement the sql statement which should be executed
+     * @param dataId the dataId of the data to be inserted
+     * @param data the data to be inserted
+     */
+    fun insertDataIntoSqlDatabase(conn: Connection?, sqlStatement: String, dataId: String, data: String): Boolean {
+        val preparedStatement: PreparedStatement?
+        var rowsInserted = 0
+        if (conn != null) {
+            preparedStatement = conn.prepareStatement(sqlStatement)
+            preparedStatement.setObject(1, UUID.fromString(dataId))
+            preparedStatement.setString(2, data)
+
+            rowsInserted = preparedStatement.executeUpdate()
+            if (rowsInserted > 0) {
+                logger.info("Data for $dataId was inserted successfully.")
+            }
+            preparedStatement?.close()
+            conn?.close()
+        }
+        return rowsInserted > 0
```

Comment:
Ist das hier eine gute Idee? Ist vorher nicht klar welche Zahl an rows erwartet wird?

Reply:
Der Gedanke ist, dass wenn das Insert nicht geklappt hat, dieser Return nicht funktioniert. Der zurückgebene Wert true/false wird dann in dem retryWrapper ausgewertet, ob das Inserting wiederholt werden muss.